### PR TITLE
[FE#89] Generic MapSelect component

### PR DESCRIPTION
**Pull request by boris-arkenaar**
closes Signalen/frontend#89

~Still contains changes from #1250. Will clear up when that PR has been merged.~

**Changes**

This PR makes a generic version of the `MapSelect` component(s) called `MapSelectGeneric`.

The following components

- `src/components/MapSelect`
- `src/signals/incident/components/IncidentPreview/components/MapSelect`
- `src/signals/incident/components/form/MapSelect`

have a generic version in their respective locations. The two `MapSelect` components in the `incident` module will always be imported instead of their generic counterpart. Inside each `MapSelect` component the choice is made to use it, or switch to the generic version instead, based on the new feature flag `useMapSelectGeneric`.

Some functionality (like custom icons and legend) has not been ported to the generic versions yet.

Added two open source icons and referenced the author in the readme.

### CODE REVIEW

**Comment by boris-arkenaar** [comment]
Upped the jest thresholds, but for functions, I had to lower it a bit. This is because of the copy of the `MapSelect` component(s).

**Review by aditudorache — CHANGES_REQUESTED** [review]

File: src/components/MapSelect/index.js [hunk 1]
```diff
@@ -64,10 +70,22 @@ const MapSelect = ({
   );
 
   const fetchRequest = useCallback(
-    bbox_str =>
-      request(`${geojsonUrl}&bbox=${bbox_str}`).catch(() => {
+    lnglatString => {
```
[the remark this answers is not in the document]
Comment (boris-arkenaar) [reply src/components/MapSelect/index.js:73]: There are now two versions of bounding box strings though. A lat-lng version, and a lng-lat version.

File: src/components/MapSelect/index.js [hunk 2]
```diff
@@ -100,8 +118,8 @@ const MapSelect = ({
            */
           pointToLayer: /* istanbul ignore next */ (feature, latlong) =>
             L.marker(latlong, {
-              icon: getIcon(feature.properties[iconField], selection.current.has(feature.properties[idField])),
-              alt: feature.properties.objectnummer,
+              icon: selection.current.has(feature.properties[idField]) ? LeafletDotSelectedIcon : LeafletDotIcon,
```
[the remark this answers is not in the document]
Comment (boris-arkenaar) [reply src/components/MapSelect/index.js:121]: Not necessary for this first iteration. No functionality is lost; the old functionality resides in `MapSelectAmsterdam`.

File: src/components/MapSelectAmsterdam/index.js [hunk 3]
```diff
@@ -0,0 +1,259 @@
+import React, { useLayoutEffect, useEffect, useState, useCallback, useMemo, useRef } from 'react';
```
[the remark this answers is not in the document]
Comment (boris-arkenaar) [reply src/components/MapSelectAmsterdam/index.js:1]: No, there is no relation. This is simply the old version of `MapSelect` exactly as it was. When we iterate further we can have all functionality in the new `MapSelect` and `MapSelectAmsterdam` could go. Did you read the description of this PR? ;)

File: src/components/MapSelect/index.js [hunk 4]
```diff
@@ -64,10 +70,22 @@ const MapSelect = ({
...
+        (acc, [key, replacement]) => acc.replace(new RegExp(`{{${key}}}`, 'g'), replacement),
+        geojsonUrl
+      );
+      return request(requestUrl).catch(() => {
         errorControl.show();
-      }),
+      });
+    },
     [errorControl, geojsonUrl]
   );
```
[the remark this answers is not in the document]
Comment (boris-arkenaar) [reply src/components/MapSelect/index.js:91]: Yes. I  moved that up to the next iteration though.

File: src/components/MapSelect/index.js [hunk 5]
```diff
@@ -229,28 +235,14 @@ const MapSelect = ({
...
 MapSelect.propTypes = {
   latlng: PropTypes.exact({
     latitude: PropTypes.number.isRequired,
     longitude: PropTypes.number.isRequired,
   }).isRequired,
   geojsonUrl: PropTypes.string.isRequired,
   onSelectionChange: PropTypes.func,
-  getIcon: PropTypes.func.isRequired,
   hasGPSControl: PropTypes.bool,
-  legend: PropTypes.arrayOf(
```
[the remark this answers is not in the document]
Comment (boris-arkenaar) [reply src/components/MapSelect/index.js:247]: Yes. Next iteration ;)

**Comment by boris-arkenaar** [comment]
See Signalen/frontend#92 for expanding `MapSelectGeneric`.

**Review by aditudorache — APPROVED** [review]
Looks good for now, further integration between the 2 MapSelect and MapSelectGeneric will follow!

**Review by MauritsR — APPROVED** [review]

**Pull request merged**